src/plugins/pwdchan/src/lib.rs

```diff
@@ -4,19 +4,24 @@ extern crate slapi_r_plugin;
 use base64;
 use openssl::{hash::MessageDigest, pkcs5::pbkdf2_hmac, rand::rand_bytes};
 use slapi_r_plugin::prelude::*;
-use std::fmt;
+use std::fmt::Write;
+use once_cell::sync::Lazy;
+use std::sync::RwLock;
+use std::convert::TryInto;
+use std::os::raw::c_char;
+
+const DEFAULT_PBKDF2_ROUNDS: usize = 10_000;
+const MIN_PBKDF2_ROUNDS: usize = 10_000;
+const MAX_PBKDF2_ROUNDS: usize = 1_000_000;
+
+const PBKDF2_ROUNDS_ATTR: &str = "nsslapd-pwdPBKDF2Rounds";
+static PBKDF2_ROUNDS: Lazy<RwLock<usize>> = Lazy::new(|| RwLock::new(DEFAULT_PBKDF2_ROUNDS));
```

Comment:
Consider using https://doc.rust-lang.org/std/sync/atomic/struct.AtomicUsize.html with Ordering::Relaxed

Reply:
I just realized we need to make the `PBKDF2_ROUNDS` a HashMap, as we might use different values in different schemes.

Do you know what the best option for our case is?

I found `dashmap` option `Lazy<DashMap<MessageDigest, usize>>` or we can make a RwLock HashMap like this `RwLock<HashMap<MessageDigest, usize>>`...
And I think there can be more...

Reply:
Okay, I tried a few approaches (even Rust's generics...), and I think the one in the last commit will work the best.
It's the simple `AtomicUsize` variables - one for each digest.